iothub/device/samples/convention-based-samples/TemperatureController/TemperatureControllerSample.cs

```diff
@@ -331,11 +347,11 @@ private async Task SendDeviceSerialNumberPropertyIfNotCurrentAsync(CancellationT
             // Retrieve the device's properties.
             ClientProperties properties = await _deviceClient.GetClientPropertiesAsync(cancellationToken);
 
-            if (!properties.TryGetValue(serialNumber, out string serialNumberReported)
+            if (!properties.Properties.TryGetValue(serialNumber, out string serialNumberReported)
```

Comment:
I think this is what you mean as I don't see the Properties property in the class definition.

```suggestion
            if (!properties.TryGetValue(serialNumber, out string serialNumberReported)
```